Add Datadog Integration Topic to Software Docs (#896)
astronomer/docs: software/integrate-datadog.md

---
sidebar_label: 'Integrate with Datadog'
title: 'Integrate Datadog with Astronomer Software'
id: integrate-datadog
description: Integrate your Datadog instance with Astronomer Software.
---

> Integrate your Datadog instance with Astronomer Software to view Airflow metrics in Datadog. These metrics can include running DAG processes, task data, pool data, and executor data. To view a complete list of the Airflow metrics collected by the Datadog Agent, see [Data Collected](https://docs.datadoghq.com/integrations/airflow/?tabs=host#data-collected). 

Reviewer: ```suggestion
Integrate your Datadog account with Astronomer Software to view Airflow metrics in Datadog. These metrics can include running DAG processes, task data, pool data, and executor data. To view a complete list of the Airflow metrics collected by the Datadog Agent, see [Data Collected](https://docs.datadoghq.com/integrations/airflow/?tabs=host#data-collected). 
```

DD is SaaS. The customer does not have an instance.

> Integrate your Datadog instance with Astronomer Software to view Airflow metrics in Datadog. These metrics can include running DAG (Directed Acyclic Graphs) processes, task data, pool data, and executor data. To view a complete list of the Airflow metrics collected by the Datadog Agent, see [Data Collected](https://docs.datadoghq.com/integrations/airflow/?tabs=host#data-collected). 

Author: @jwitz force of habit. As discussed previously, acronyms are always defined on first use. If the Astronomer standard is not to define _DAG_ on first use, it should be noted in the style guide.

1. Install the Datadog Agent on your Kubernetes cluster. See [Install the Datadog Agent on Kubernetes](https://docs.datadoghq.com/containers/kubernetes/installation/?tabs=operator).

> 2. Open the `values.yaml` file and add this entry:

Reviewer 2: ```suggestion
2. Open the [Datadog `values.yaml` configuration file](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml) and add a new configuration under `datadog.confd.openmetrics.instances` for each Deployment you want to integrate with Datadog:
```

I know that Astronomer's config file is called `config.yaml`, but it might still be helpful to distinguish this as the Datadog helm chart.

Reviewer 2: This is a bit of an awkward phrasing because we tell customers to "Integrate Datadog with Software", but the actual integration is on a per-Deployment basis. Maybe it makes more sense to frame this doc in the context of integrating a Deployment instead of the entire platform? Your call

>     ```yaml
>     datadog:  
>       confd:
>         openmetrics.yaml: |-
>           instances:
>             # The airflow statsd endpoint to query from
>             - prometheus_url: http://<airflow-release-name>-statsd.<airflow-release-namespace>.svc:9102/metrics
>               namespace: "<airflow-release-namespace>"
>               metrics:
>                 - "*"
>    ```

Reviewer 2: ```suggestion
    ```yaml
    datadog:  
      confd:
        openmetrics.yaml: |-
          instances:
            # The airflow statsd endpoint to query from
            - prometheus_url: http://<deployment-release-name>-statsd.<platform-namespace>.svc:9102/metrics
              namespace: "<platform-namespace>"
              metrics:
                - "*"
    ```
```

Reviewer 2: I'm pretty sure this is what these values are referenced as in our public docs, with the Deployment release name being for Deployments only (https://docs.astronomer.io/software/configure-deployment#customize-release-names) and the release name being where the entire Astronomer platform is installed (we recommend customers use `astronomer` for this during their install).

Reviewer 2: Also, is it worth telling people how to retrieve their Deployment release names here? It's available in the Software UI under settings:
![Screen Shot 2022-07-08 at 11 28 16 AM](https://user-images.githubusercontent.com/74574233/178041323-92622043-6836-45ff-a6ea-6ba33adaae9e.png)

Come to think of it I should probably document this somewhere other than the CLI reference guide, which is the only place that content is available 😬

Author: @jwitz I added this statement below the procedure step and command example:

_To locate the Deployment release name, select a Deployment in Astronomer Software and copy the value displayed in the **Release Name** field._

> 3. Optional. Add additional entries for your other Airflow instances. For example:

Reviewer 2: If we frame this doc in the context of integrating a single Deployment, I agree that this is optional. But a Software user is likely to get all of this configuration done at once and "forget it" if they're configuring at a platform level, in which case configuring multiple Deployments would not be optional. 

My preference would be to frame the setup as "do this for each Deployment you want to integrate with Datadog", meaning that we would only have one non-optional example of how to add multiple Deployments to the configuration. This is likely how Software admins will use this feature.

Author: @jwitz I disagree. I think the procedure is presented correctly in its current form. If Software users want to configure multiple Deployments at once, I think it's clear how this can be accomplished. We're both making assumptions about the audience. My experience tells me this is the better option. If you feel strongly about a change, we can discuss it.

    ```yaml
    datadog:  
      confd:
    openmetrics.yaml: |-
      instances:

>         # The airflow statsd endpoint to query from
>         - prometheus_url: http://<airflow-release-name>-statsd.<airflow-release-namespace>.svc:9102/metrics
>           namespace: "<airflow-release-namespace>"
>           metrics:
>             - "*"
>         - prometheus_url: http://<airflow-release-name>-statsd.<airflow-release-namespace>.svc:9102/metrics
>           namespace: "<airflow-release-namespace>"

Reviewer 2: ```suggestion
        # The airflow statsd endpoint to query from
        - prometheus_url: http://<first-deployment-release-name>-statsd.<platform-namespace>.svc:9102/metrics
          namespace: "<platform-namespace>"
          metrics:
            - "*"
        - prometheus_url: http://<second-deployment-release-name>-statsd.<platform-namespace>.svc:9102/metrics
          namespace: "<platform-namespace>"
```

          metrics:  
            - "*"
    ```
4. Save your changes to the `values.yaml` file and then run the following command to update your Datadog configuration:

    ```shell
    helm upgrade datadog -f datadog-values.yaml --set datadog.site='datadoghq.com' --set datadog.apiKey=<my-api-key> datadog/datadog -n datadog
    ```

> 5. Optional. Run the following command to confirm your confuguration changes were implemented:

Reviewer: ```suggestion
5. Optional. Run the following command to confirm your configuration changes were implemented:
```

> 5. Optional. Run the following command to confirm your configuration changes were implemented:

Reviewer: ```suggestion
5. Optional: Confirming your configuration changes were implemented. First find the Datadog agent pod using `kubectl get pods -n datadog`, and then use the run the following command:
```

The pod name below is dynamic

    ```bash

>     kubectl exec -it dd-test-datadog-f4vhx -n astronomer -- bash -c 'agent status' Defaulting container name to agent.

Reviewer: ```suggestion
    kubectl exec -it <dd-test-datadog pod name> -n astronomer -- bash -c 'agent status' Defaulting container name to agent.
```

Pod name is dynamic

Reviewer: @rbankston is `dd-test` templated from the release name??
